src/rendering/vcTileRenderer.cpp

```diff
@@ -223,15 +223,17 @@ uint32_t vcTileRenderer_LoadThread(void *pThreadData)
       }
 
       udResult result = udR_Failure_;
-      void *pFileData = nullptr;
-      int64_t fileLen = -1;
+      uint32_t amountRead = 0;
       int width = 0;
       int height = 0;
       int channelCount = 0;
       uint8_t *pData = nullptr;
+      udFile *pFile = nullptr;
+      char buffer[20000] = {};
```

Comment:
Some tiles might be bigger than this especially if they are higher res than 256² or full colour satellite- they could be much larger.

@samsinsane and I had an offline discussion that the best way to do this might be for file handlers to be a bit more flexible.

After an additional discussion with @dpev, he suggested the simplest/cleanest option would be if file-handlers could provide an (optional) load file function pointer; current behaviour continues if that is nullptr, otherwise it uses that function to load the whole file.